Create index.md (#2032)
github/explore: topics/ui/index.md

---
aliases: user-interface

> created_by: Glenda Schroeder

Reviewer: This field is meant to be used for the creator of the subject of the topic, in this case `ui`. I do not see `Glenda Schroeder` anywhere on the wikipedia page _and_ I think this topic is too broad to have any `created_by` attribute. Can we remove this altogether?
```suggestion
```

display_name: User interface

> released: '1985'

Reviewer: In 1985 the `G`raphical `U`ser `I`nterface was introduced, but I think that `UI` was probably introduced long before. I think this is another instance of this topic being too broad to include a specific year like this, thoughts on removing it?
```suggestion
```

> short_description: A UI is what users use to control a program through a graphical interface

Reviewer: The qualification of "through a graphical interface" seems misleading here. That should be the territory of `GUI`. Wikipedia has a nice succinct description we could use here:
```suggestion
short_description: A user interface (UI) is the space where interactions between humans and machines occur.
```

topic: ui
wikipedia_url: https://en.wikipedia.org/wiki/User_interface
---

> The goal of user interface design is to produce a user interface which makes it easy, efficient, and enjoyable (user-friendly) to operate a machine in the way which produces the desired result (i.e. maximum usability). This generally means that the operator needs to provide minimal input to achieve the desired output, and also that the machine minimizes undesired outputs to the user.

Reviewer: This description's subject seems like it might be for "User Interface Design" instead of "User Interface". Should this topic addition be `ui-design`?

Reviewer: This comment still seems relevant, any thoughts to changing this topic to `ui-design`?

Author: > This comment still seems relevant, any thoughts to changing this topic to `ui-design`?

I'm not really sure how I feel. I feel like `ui-design` is like a part of `ui` if that makes any sense?

Reviewer: The conflict I see is that we're targeting UI in all the other places except the long description, see here for the wikipedia article on UI design, which is different than just UI: https://en.wikipedia.org/wiki/User_interface_design

Author: ah i see, then i say we change it to `ui-design` and i'll make another, more generalized pr about `ui` later on.
